pkg/config/cmd/cmd.go

```diff
@@ -4,23 +4,74 @@ import (
 	"fmt"
 	"io"
 
+	"github.com/spf13/pflag"
+
 	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/meta"
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/labels"
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
+type Runner interface {
+	Run(list *kapi.List, namespace string) []error
+}
+
 // AfterFunc takes an info and an error, and returns true if processing should stop.
 type AfterFunc func(*resource.Info, error) bool
 
+// OpFunc takes the provided info and attempts to perform the operation
+type OpFunc func(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error)
+
 // Bulk provides helpers for iterating over a list of items
 type Bulk struct {
-	Mapper            meta.RESTMapper
-	Typer             runtime.ObjectTyper
-	RESTClientFactory func(mapping *meta.RESTMapping) (resource.RESTClient, error)
-	After             AfterFunc
-	Retry             func(info *resource.Info, err error) runtime.Object
+	Mapper resource.Mapper
+
+	Op    OpFunc
+	After AfterFunc
+	Retry func(info *resource.Info, err error) runtime.Object
+}
+
+// Create attempts to create each item generically, gathering all errors in the
+// event a failure occurs. The contents of list will be updated to include the
+// version from the server.
+func (b *Bulk) Run(list *kapi.List, namespace string) []error {
```

Comment:
I checked and I got 100% coverage of branches via test-cmd.sh :)  But yes,
I'll add some.

On Mon, Apr 18, 2016 at 8:31 AM, David Eads <notifications@github.com>
wrote:

> In pkg/config/cmd/cmd.go
> <https://github.com/openshift/origin/pull/8539#discussion_r60049141>:
>
> > -	Mapper            meta.RESTMapper
> > -	Typer             runtime.ObjectTyper
> > -	RESTClientFactory func(mapping *meta.RESTMapping) (resource.RESTClient, error)
> > -	After             AfterFunc
> > -	Retry             func(info *resource.Info, err error) runtime.Object
> > +	Mapper resource.Mapper
> > +
> > +	Op    OpFunc
> > +	After AfterFunc
> > +	Retry func(info *resource.Info, err error) runtime.Object
> > +}
> > +
> > +// Create attempts to create each item generically, gathering all errors in the
> > +// event a failure occurs. The contents of list will be updated to include the
> > +// version from the server.
> > +func (b *Bulk) Run(list *kapi.List, namespace string) []error {
>
> If we're going to collapse onto this, how about a few tests?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8539/files/e44ca5593b4d8049f941c4573a5a245248f6481a#r60049141>
>